### PR TITLE
Add ordering in matrix 2d form

### PR DESCRIPTION
**Pull request by AdityaKhatri**
- Addresses https://github.com/the-deep/client/issues/1806
- Depends: https://github.com/the-deep/deep-ui/pull/67

## Changes

* Add ordering to Matrix2D

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

>NOTE: 4 new TS issues have been introduced which will be solved after new DEEP UI is released

### CODE REVIEW

File: src/newViews/AnalyticalFramework/Canvas/index.tsx [hunk 1]
```diff
@@ -179,9 +179,9 @@ function Canvas<T>(props: Props<T>) {
         isSecondary,
         widget: data,
         onWidgetValueChange: handleWidgetValueChange,
-        showWidgetEdit: props.editMode,
+        showWidgetEdit: !props.editMode,
         onWidgetEditClick: handleWidgetEditClick,
-        showWidgetDelete: props.editMode,
+        showWidgetDelete: !props.editMode,
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/Canvas/index.tsx:184]: Why negate these values?

File: package.json [hunk 2]
```diff
@@ -127,13 +127,13 @@
...
     "dependencies": {
-        "@hcaptcha/react-hcaptcha": "^0.3.6",
         "@dnd-kit/core": "^3.1.1",
         "@dnd-kit/modifiers": "^3.0.0",
         "@dnd-kit/sortable": "^4.0.0",
         "@dnd-kit/utilities": "^2.0.0",
+        "@hcaptcha/react-hcaptcha": "^0.3.6",
         "@mapbox/mapbox-gl-draw": "^1.2.0",
-        "@the-deep/deep-ui": "1.4.2-beta.0",
+        "@the-deep/deep-ui": "^1.4.2",
```
Comment (subinasr) [inline package.json]: What does the '^' mean?

File: src/utils/safeCommon.ts [hunk 3]
```diff
@@ -12,6 +12,10 @@ export function compareOrder<T extends { order: number }>(a: T, b: T) {
     return compareNumber(a.order, b.order);
 }
 
+export function reOrder<T extends { order: number }>(data: T[]) {
```
Comment (tnagorra) [inline src/utils/safeCommon.ts]:
```suggestion
export function reorder<T extends { order: number }>(data: T[]) {
```

Comment (tnagorra) [reply src/utils/safeCommon.ts]:
```suggestion
export function reOrder<T extends { order: number }>(data: T[]): T[] {
```

File: src/newViews/AnalyticalFramework/WidgetPreview/Matrix2dWidgetInput/index.tsx [hunk 4]
```diff
@@ -283,6 +291,10 @@ function Matrix2dWidgetInput<N extends string>(props: Props<N>) {
         [disabled, readOnly, handleSubRowChange, value, columns],
     );
 
+    const orderedRows = useMemo(() => (
+        widget?.data?.rows?.sort(compareOrder)
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/WidgetPreview/Matrix2dWidgetInput/index.tsx]: Please make a copy before sorting

File: src/newViews/AnalyticalFramework/WidgetPreview/Matrix2dWidgetInput/index.tsx [hunk 5]
```diff
@@ -270,7 +275,10 @@ function Matrix2dWidgetInput<N extends string>(props: Props<N>) {
         [],
     );
 
-    const columns = widget?.data?.columns;
+    const columns = useMemo(() => (
+        widget?.data?.columns?.sort(compareOrder)
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/WidgetPreview/Matrix2dWidgetInput/index.tsx]: Please make a copy before sorting

File: src/newViews/AnalyticalFramework/WidgetPreview/Matrix2dWidgetInput/index.tsx [hunk 6]
```diff
@@ -165,6 +166,10 @@ function Row(props: RowProps) {
         subRows,
     } = row;
 
+    const orderedSubRows = useMemo(() => (
+        subRows?.sort(compareOrder)
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/WidgetPreview/Matrix2dWidgetInput/index.tsx]: Please make a copy before sorting

File: src/newViews/AnalyticalFramework/Canvas/index.tsx [hunk 7]
```diff
@@ -179,9 +179,9 @@ function Canvas<T>(props: Props<T>) {
         isSecondary,
         widget: data,
         onWidgetValueChange: handleWidgetValueChange,
-        showWidgetEdit: props.editMode,
+        showWidgetEdit: !props.editMode,
```
Comment (subinasr) [inline src/newViews/AnalyticalFramework/Canvas/index.tsx:182]: I maybe wrong, but doesn't this mean, widget should be editable when edit mode is false? Or is it that when editMode is false it means the mode is 'edit mode'? If so, isn't it more practical to change the default value?

**Review by subinasr — COMMENTED** [review]
Not requesting changes, but please answer my questions.

File: package.json (hunk 2 above)
Comment (AdityaKhatri) [reply package.json]: ~version “Approximately equivalent to version”, will update you to all future patch versions, without incrementing the minor version. ~1.2.3 will use releases from 1.2.3 to <1.3.0.

^version “Compatible with version”, will update you to all future minor/patch versions, without incrementing the major version. ^2.3.4 will use releases from 2.3.4 to <3.0.0.

File: src/newViews/AnalyticalFramework/Canvas/index.tsx (hunk 7 above)
Comment (AdityaKhatri) [reply src/newViews/AnalyticalFramework/Canvas/index.tsx:182]: Yes, editMode is for when a widget is currently being edited.
If any of the widgets are being edited, the edit buttons should be hidden.

**Review by tnagorra — APPROVED** [review]

**Pull request merged**